File: src/main/java/com/protryon/jasm/stage1/constant/CString.java
package com.protryon.jasm.stage1.constant;

import com.protryon.jasm.stage1.CConstant;

import java.io.DataOutputStream;
import java.io.IOException;

public class CString extends CConstant<Integer> {
	public CString(int ref) {
		super(ref, 8);
	}

	protected void write(DataOutputStream out) throws IOException {
		out.writeShort(this.value);
	}
}

File: src/main/java/com/protryon/jasm/stage1/constant/CString.kt
package com.protryon.jasm.stage1.constant

import com.protryon.jasm.stage1.CConstant

import java.io.DataOutputStream
import java.io.IOException

class CString(ref: Int) : CConstant<Int>(ref, 8) {

    @Throws(IOException::class)
    override fun write(out: DataOutputStream) {
        out.writeShort(this.value)
    }
}
